python/cudf/cudf/core/indexed_frame.py

```diff
@@ -1324,6 +1324,8 @@ def _drop_na_rows(
         thresh: int, optional
             If specified, then drops every row containing
             less than `thresh` non-null values.
+        drop_nan: bool
+            `nan` is also considered as `NA`
```

Comment:
```suggestion
        drop_nan : bool, optional
            If True, `NaN` values are also dropped.
```

Reply:
From what I can see, the `drop_nan` flag should not exist at all.

`drop_nan` defaults to `False` but the function is only called in one place. In that call, `drop_nan` is hard-coded as `True`. https://github.com/rapidsai/cudf/blob/baff5cfd4cb9deb8f7887d597740c0728cbd4407/python/cudf/cudf/core/indexed_frame.py#L1301-L1303

Moreover, the corresponding method `_drop_na_columns` does not have such a flag. https://github.com/rapidsai/cudf/blob/baff5cfd4cb9deb8f7887d597740c0728cbd4407/python/cudf/cudf/core/frame.py#L1242

Reply:
When `drop_nan` is True, it doesn't always drop that row, it only makes it consider the `nan` row as `NA`. Only when the criteria of the row meets `any`/`all` the row is dropped.

Reply:
Currently it's depended by an thrid party user. As a hotfix PR it should focus on fixing existing broken workflow and defer its usage/dependency question to the next release.

Reply:
Users cannot depend on the functionality of internal APIs. I understand the goal of avoiding breakage, but if we defer we'll need to make a follow-up PR for the next release.

Reply:
I created #10125 to track this. 👍

Reply:
After this PR is merged into 22.02 and forward-merged into 22.04, I'll work on this fix.